### PR TITLE
SWC-6653

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/ACTAccessRequirementWidget.java [hunk 1]
```diff
@@ -107,6 +109,10 @@ public void invoke() {
         refreshApprovalState();
       }
     };
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/ACTAccessRequirementWidget.java:111]: Expose the AR ID for the legacy ACT AR widget, and the Lock AR (if ACT)

Comment (jay-hodgson) [reply src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/ACTAccessRequirementWidget.java:111]: And only show the "Covered Entities" if ACT

File: src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/AccessRequirementRelatedProjectsList.java [hunk 2]
```diff
@@ -15,22 +16,26 @@ public class AccessRequirementRelatedProjectsList implements IsWidget {
...
     this.isACTMemberAsyncHandler = isACTMemberAsyncHandler;
+    this.propsProvider = propsProvider;
     container.setVisible(false);
   }
 
   public void configure(String accessRequirementId) {
     AccessRequirementRelatedProjectsListProps props =
       AccessRequirementRelatedProjectsListProps.create(accessRequirementId);
-    ReactNode component = React.createElementWithThemeContext(
+    ReactNode component = React.createElementWithSynapseContext(
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/AccessRequirementRelatedProjectsList.java:35]: Bug 2:  Pass the full Synapse Context, including the access token.  Without this, only Public projects were able to be resolved in the related projects component.

File: src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/LockAccessRequirementWidgetViewImpl.java [hunk 3]
```diff
@@ -55,7 +65,7 @@ public void setTeamSubjectsWidget(IsWidget w) {
   }
 
   @Override
-  public void setEntitySubjectsWidget(IsWidget entitySubjectsWidget) {
+  public void setEntitySubjectsWidget(IsWidget w) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/LockAccessRequirementWidgetViewImpl.java:68]: Bug 1:  Typo.  `entitySubjectsWidget `should be `w `(local vs member variable)

**Comment by jay-hodgson** [comment]
As ACT:
With fix for bug 1 (lock AR rendering issue):
Screen Shot 2024-01-11 at 1 42 28 PM

Additionally, with fix for bug 2 (context passed to related projects):
Screen Shot 2024-01-11 at 2 08 46 PM

As non-ACT:
Screen Shot 2024-01-11 at 2 10 35 PM

File: src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/AccessRequirementRelatedProjectsList.java (hunk 2 above)
Comment (hallieswan) [reply src/main/java/org/sagebionetworks/web/client/widget/accessrequirements/AccessRequirementRelatedProjectsList.java:35]: Thanks for catching this!

**Review by nickgros — APPROVED** [review]

**Review by hallieswan — APPROVED** [review]

**Pull request merged**